shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -473,6 +485,63 @@ public boolean performEditorAction(int actionCode) {
     return true;
   }
 
+  @Override
+  public boolean commitContent(InputContentInfo inputContentInfo, int flags, Bundle opts) {
+      // Ensure permission is granted
+      if (BuildCompat.isAtLeastNMR1() && (flags & InputConnectionCompat.INPUT_CONTENT_GRANT_READ_URI_PERMISSION) != 0) {
+        try {
+            inputContentInfo.requestPermission();
+        } catch (Exception e) {
+            return false;
+        }
+      }
+
+      if (inputContentInfo.getDescription().getMimeTypeCount() > 0) {
+        inputContentInfo.requestPermission();
+
+        final Uri uri = inputContentInfo.getContentUri();
+        final String mimeType = inputContentInfo.getDescription().getMimeType(0);
+        Context context = mFlutterView.getContext();
+        Boolean retval = false;
+
+        try {
+          final InputStream is = context.getContentResolver().openInputStream(uri);
+          final byte[] data = this.readStreamFully(is, 64 * 1024);
+
+          final Map<String, Object> obj = new HashMap<>();
+          obj.put("mimeType", mimeType);
+          obj.put("data", data);
+          obj.put("uri", uri != null ? uri.toString() : null);
+
+          // Commit the content to the text input channel
+          textInputChannel.commitContent(mClient, obj);
+          retval = true;
+        } catch (FileNotFoundException ex) {
+        } catch (Exception ex) {
```

Comment:
you might want to split the try block into two, so it's more clear what line might throw.  Also, you can `return false;`,  in the catch block, remove `retval`, and return true in line 518.